app/views/items/show.html.erb

```diff
@@ -17,44 +17,46 @@
       <% else %>
         <p style= "color:red">Inactive</p>
       <% end %>
-      <p><%= link_to "Edit Item", "/items/#{@item.id}/edit" %></p>
-      <p><%= link_to "Delete Item", "/items/#{@item.id}", method: :delete if @item.no_orders?%></p>
       <%= link_to "Add Review", "/items/#{@item.id}/reviews/new" %>
+      <% if current_merchant? || current_admin? %>
+        <p><%= link_to "Delete Item", "/items/#{@item.id}", method: :delete if @item.no_orders?%></p>
+        <p><%= link_to "Edit Item", "/items/#{@item.id}/edit" %></p>
+      <% end %>
       <% if !current_admin? %>
```

Comment:
This is a good catch and it makes sense that only an admin and merchant can do those things.